File: c#/tema_5/Persistence/IRepository.cs
using System.Collections.Generic;
using Model;

namespace Persistence
{
    public interface IRepository<T, TId> where T : IEntity<TId>
    {
        T FindOne(TId id);
        IEnumerable<T> FindAll();
        void Save(T entity);
        void Update(T entity, TId id);
        void Delete(TId id);
    }
}

File: java/tema_5/ConcertProPersistance/src/main/java/concertpro/persistance/repository/IRepository.java
package concertpro.persistance.repository;

import concertpro.model.Entity;

public interface IRepository<T extends Entity<ID>, ID> {

    T findOne(ID id);
    Iterable<T> findAll();
    void save(T elem);
    void update(T elem, ID id);
    void delete(ID id);

}
